icecreamshop/src/test/java/com/summerschool/icecreamshop/controller/RateControllerTest.java

```diff
@@ -49,6 +51,17 @@ public void setUp(){
         rateDTO.setRate(4);
     }
 
+    @Test
+    public void testCreateRate() {
+
+        Mockito.when(modelMapper.map(rateDTO, Rate.class)).thenReturn(rate);
+        Mockito.when(rateService.add(rate)).thenReturn(rate);
+
+        ResponseEntity<RateDTO> r = rateController.add(rateDTO);
+
+        assertEquals(HttpStatus.CREATED, r.getStatusCode());
+    }
+
```

Comment:
ai putea adauga un test pentru cazul in care  ai Mockito.when(rateService.add(rate)).thenReturn(null);
si sa ai apoi assertEquals(422_UnprocessableEntity, r.getStatusCode())